src/Files.App/Views/Layouts/BaseLayoutPage.cs

```diff
@@ -702,8 +702,8 @@ public void UpdateSelectionSize()
 			var isSizeKnown = !items.Any(item => string.IsNullOrEmpty(item.FileSize));
 			if (isSizeKnown)
 			{
-				long size = items.Sum(item => item.FileSizeBytes);
-				SelectedItemsPropertiesViewModel.ItemSizeBytes = size;
+				ulong size = items.Aggregate(0UL, (current, item) => current + (ulong)item.FileSizeBytes);
+				SelectedItemsPropertiesViewModel.ItemSizeBytes = (long)size;
```

Comment:
Is casting here ok when overflow exception is happening?

Reply:
I'd suggest 
```suggestion
				SelectedItemsPropertiesViewModel.ItemSizeBytes = size >= (ulong)long.MaxValue ? -1l : (long)size; // NOTE: Suppressed OverflowException in #16165
```
What do you think?

Reply:
Fyi @hishitetsu

Reply:
OverflowException may occur even in ulong.
How about using decimal?

Reply:
> OverflowException may occur even in ulong.

True, but it's already very uncommon with long (2 reports) and switching to ulong should take care of it.

Reply:
Does `ToSizeString` work with decimal?

Reply:
Hmm, in the first place WIN32_FIND_DATA, which we use for enumeration of items, can only retrieve up to sizeof(ulong)

Reply:
Compromise may be necessary.

Btw, should also be updated to ulong.
https://github.com/files-community/Files/blob/8ba4f4517a35e85547a89ec63a08805e00f60965/src/Files.App/Data/Items/ListedItem.cs#L281

Reply:
> Does `ToSizeString` work with decimal?

We can cast decimal to double.

Reply:
Casting decimal to long may throw an OverflowException.

Reply:
What's the reason to do so?
If I understand C#'s value types correctly, ulong can express larger value (20 digits) than double (16 digits). decimal can express much larger (like 29?) than ulong or double though.

Reply:
You are right regarding the first part. But isn't the significant number the same even when it comes to integers? Like 1.00 == 100

The max value that can be expressed accurately (according to an article):
Type|Max integer
---|--:
double|`0x20000000000000`
ulong|`0xFFFFFFFFFFFFFFFF`

double.MaxValue is 0x7FEFFFFFFFFFFFFF so even this is less than ulong.MaxValue...

But you seem to understand much better, I'm fine to merge.

Reply:
You are right 😄
How large is that digits?

Reply:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/floating-point-numeric-types

> ±5.0 × 10^−324 to ±1.7 × 10^308

Reply:
@hishitetsu can you check my latest commit?

Reply:
Actually, I haven't been able to check it out yet. I think I will be able to see it tomorrow morning in Japan time.